src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/commands/rc_command_request.h

```diff
@@ -0,0 +1,151 @@
+namespace commands {
+
+class RCCommandRequest : public app_mngr::commands::CommandRequestImpl {
+ public:
+  /**
+   * @brief RCCommandRequest class constructor
+   *
+   **/
+  RCCommandRequest(
+      rc_rpc_plugin::ResourceAllocationManager& resource_allocation_manager,
+      const app_mngr::commands::MessageSharedPtr& message,
+      app_mngr::ApplicationManager& application_manager);
+
+  virtual ~RCCommandRequest();
+
+  virtual void OnTimeout();
+
+  void Run();
+
+  virtual void on_event(const app_mngr::event_engine::Event& event);
+
+ protected:
+  ResourceAllocationManager& resource_allocation_manager_;
+  bool auto_allowed_;
+
+  /**
+   * @brief AcquireResource try to allocate resource for application
+   * In case if allocation of resource is not required, return ALLOWED by
+   * default.
+   * This method should be overrided in RPCs that requires resource allocation
+   * @return result of resource allocation, in case if allocation os not
+   * required, return ALLOWED
+   */
+  virtual AcquireResult::eType AcquireResource(
+      const app_mngr::commands::MessageSharedPtr& message) {
+    return AcquireResult::ALLOWED;
+  }
+
+  /**
+   * @brief IsResourceFree check resource state
+   * This is default implementation which has to be redefined for RPCs which
+   * need to manage the resources
+   * @param module_type Resource name
+   * @return True if free, otherwise - false
+   */
+  virtual bool IsResourceFree(const std::string& module_type) const {
+    UNUSED(module_type);
+    return true;
+  }
+  /**
+   * @brief SetResourceState changes state of resource
+   * This is default implementation which has to be redefined for RPCs which
+   * need to manage the resources
+   * @param Message containing type of module to extract
+   * @param State to set for resource
+   */
+  virtual void SetResourceState(const std::string& module_type,
+                                const ResourceState::eType) {}
+
+
+  TypeAccess CheckModule(const std::string& module_type,
+                         application_manager::ApplicationSharedPtr app);
+
+  bool auto_allowed() const {
+    return auto_allowed_;
+  }
+
+  void set_auto_allowed(bool value) {
+    auto_allowed_ = value;
+  }
+
+  /**
+   * @brief executes specific logic of children classes
+   */
+  void virtual Execute() = 0;
+
+  void set_disallowed_info(const std::string& info) {
+    disallowed_info_ = info;
+  }
+
+ private:
+  /**
+   * @brief CheckDriverConsent checks driver consent defined in policy table
+   * @return True if no consent is required, otherwise - false
+   */
+  bool CheckDriverConsent();
+
+  /**
+   * @brief AcquireResources checks whether resource status is busy or not and
+   * then tries to acquire this resource. In case driver consent is required -
+   * sends consent request to HMI.
+   * @return True in case of resource is free and successfully acquired,
+   * otherwise false
+   */
+  bool AcquireResources();
+  void SendDisallowed(TypeAccess access);
```

Comment:
@LitvinenkoIra TypeAccess can be kDisallowed or kAllowed.
Can this function send SendDisallowed(kAllowed) ? Confusing.